src/test/java/com/karankumar/bookproject/backend/service/UserServiceTest.java

```diff
@@ -31,36 +30,27 @@
 
 import javax.validation.ConstraintViolationException;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.karankumar.bookproject.utils.SecurityTestUtils.*;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 @IntegrationTest
+@Transactional
 class UserServiceTest {
-    private final UserService userService;
-    private final UserRepository userRepository;
-    private final RoleRepository roleRepository;
+    @Autowired private UserService userService;
+    @Autowired private UserRepository userRepository;
+    @Autowired private RoleRepository roleRepository;
 
     private final User validUser = User.builder()
                                        .username("validUser")
                                        .email("valid@testmail.com")
                                        .password("aaaaAAAA1234@")
                                        .build();
 
-    @Autowired
-    UserServiceTest(UserService userService,
-                    UserRepository userRepository,
-                    RoleRepository roleRepository) {
-        this.userService = userService;
-        this.userRepository = userRepository;
-        this.roleRepository = roleRepository;
-    }
-
-    @AfterEach
-    void tearDown() {
-        userRepository.deleteAll();
-        roleRepository.deleteAll();
-    }
-
```

Comment:
Could you please undo this change?

Reply:
Thanks for bringing back constructor injection. When you're ready for a review, please also ensure `tearDown()` is brought back

Reply:
I annotated this test with `@Transactional` so `tearDown()` is not needed anymore because the transaction is rolled back after each test.